docs/constrained-generators/manual.md

```diff
@@ -121,7 +209,156 @@ dot to the end of the Haskell name, we add an underscore `(_) to the end of the
 (member_ :: HasSpec a => Term a -> Term (Set a) -> Term Bool)
 ```
 
-# How we solve the constraints
+While the underscored function symbols, may appear to be just to an Applicative lifting over `Term`, that is not the case.
+An Applicative lifting would allow the base function to be applied under the `Term` type, but The underscored function symbols
+also know how to reason logically about the function.
+
+
+## Predefined HasSpec instances and their function symbols.
+
+In order to write specifications for a particular type, that type must have a `HasSpec` instance. 
+A type with a `HasSpec` instance might have a number of Function Symbols that operate on that type.
+There are a number of types that have predefined `HasSpec` instances. As a reference, we list them 
+here along with the types of their function symbols.
+
+### Function symbols for numeric types
+
+`(Int, Integer, Natural, Int8, Int16, Int32, Int64, Word8, Word16, Word32, Word64)`
+
+The function symbols of numeric types are:
+
+ 1. `(<=.) :: OrdLike a => Term a -> Term a -> Term Bool`
+ 2. `(<.) :: OrdLike a => Term a -> Term a -> Term Bool`
+ 3. `(>=.) :: OrdLike a => Term a -> Term a -> Term Bool`
+ 4. `(>.) :: OrdLike a => Term a -> Term a -> Term Bool`
+ 5. `(==.) :: (Eq a, HasSpec a) => Term a -> Term a -> Term Bool`
+ 6.  A partial Num instance for (Term n) where n is a Numeric type. Operators `(+)`, `(-)`, `(*)`
+ 
+### Function symbols for Bool
+
+The function symbols of `Bool` are:
+
+  1.  `or_ :: Term Bool -> Term Bool -> Term Bool`
+  2.  `not_ :: Term Bool -> Term Bool`
+ 
+###  Function symbols for List
+
+`HasSpec a => HasSpec [a]`
+
+The function symbols of `[a]` are:
+
+  1.  `foldMap_ :: (Sized [a], Foldy b, HasSpec a) => (Term a -> Term b) -> Term [a] -> Term b`
+  2.  `singletonList_ :: (Sized [a], HasSpec a) => Term a -> Term [a]`
+  3.  `append_  :: (Sized [a], HasSpec a) => Term [a] -> Term [a] -> Term [a]`
+  
+###  Function symbols for Set
+
+`HasSpec a => HasSpec (Set a)`
+
+The function symbols of `(Set a)` are:
+
+  1.  `singleton_ :: (Ord a, HasSpec a) => Term a -> Term (Set a)`
+  2.  `union_ :: (Ord a, HasSpec a) => Term (Set a) -> Term (Set a) -> Term (Set a)`
+  3.  `subset_ :: (Ord a, HasSpec a) => Term (Set a) -> Term (Set a) -> Term Bool`
+  4.  `member_ :: (Ord a, HasSpec a) => Term a -> Term (Set a) -> Term Bool`
+  5.  `disjoint_ :: (Ord a, HasSpec a) => Term (Set a) -> Term (Set a) -> Term Bool`
+  6.  `fromList_ :: (Ord a, HasSpec a) => Term [a] -> Term (Set a)`
+
+###  Function symbols for Map
+
+`(HasSpec k, HasSpec v) => HasSpec (Map k v)`
+
+The function symbols of `(Map k v)` are:
+
+1.  `dom_ :: Ord k => Term (Map k v) -> Term (Set k)`
+2.  `rng_ :: Ord k => Term (Map k v) -> Term [v]`
```

Comment:
Maybe we should rename `dom_` and `rng_` to match whatever the functions are called in `Data.Map`. Also, it should be possible to add `toSet_ :: Ord k, Ord v => Term (Map k v) -> Term (Set (k, v))` if I'm not mistaken.